axiom/nr_txn.c

```diff
@@ -3293,28 +3292,123 @@ bool nr_txn_log_decorating_enabled(nrtxn_t* txn) {
   return false;
 }
 
+#define ENSURE_LOG_LEVEL_NAME(level_name) \
+  (nr_strempty(level_name) ? "UNKNOWN" : level_name)
```

Comment:
Is it possible to get a non-NULL non-empty level_name here that does not correspond to a valid log level?

Reply:
It should not be possible under normal circumstances. But whatever the agent gets from instrumented method, it honors.